tools/itsg33-to-gh-issues/tests/test_script.py

```diff
@@ -0,0 +1,81 @@
+from unittest.mock import patch,Mock
+import pytest
+import script
+import secrets
+
+
+@patch("script.get_controls")
+@patch("script.post_request")
+def test_main(mock_post_request, mock_get_controls):
+    mock_get_controls.return_value = [
+        ["Family1", "Control1", "E1", "Control1 name", "Class1", "Definition1", "Supplemental guidance1", "References1", "IT Security Function", "", "", "", "", "", "Medium", "", "", ""],
+        ["Family2", "Control2", "E2", "Control2 name", "Class2", "Definition2", "Supplemental guidance2", "References2", "", "IT Operation Group", "", "", "", "", "Low", "", "", ""]
+    ]
+    mock_post_request.return_value.status_code = 201
+
+    script.main()
+
+    assert mock_post_request.call_count == 2
```

Comment:
If you want you can also assert exactly what the 2 calls were as well:
https://github.com/cds-snc/github-secret-scanning/blob/1c06bb43b55129301a3cbe6347938c160114929b/api/tests/routers/test_alerts.py#L31-L42

Depends on how specific you need to be for your test.

Reply:
Since the actual purpose of that part of code is to call the post_request for every control row we get, I think the actual test to see the call would be inside that post_request function. If I was calling straight there requests.post, I would test at that place. Thoughts?

Reply:
It's more just to see that when you call `post_request` it has the values you expect from each iteration of the `for` loop, which would be based on the 3 mocked functions you're calling:
```python
    mock_post_request.assert_has_calls(
        [
            call(
                "https://api.github.com/repos/owner/repo/issues",
                {
                    "Accept": "application/vnd.github+json",
                    "Authorization": "Bearer token",
                    "X-GitHub-Api-Version": "2022-11-28"
                },
                {
                    "title": "title", "body": "body", "labels": "labels"
                }
            ),
            call(
                "https://api.github.com/repos/owner/repo/issues",
                {
                    "Accept": "application/vnd.github+json",
                    "Authorization": "Bearer token",
                    "X-GitHub-Api-Version": "2022-11-28"
                },
                {
                    "title": "title", "body": "body", "labels": "labels"
                }
            ),
        ]
    )
```
Looking at the above, I'm not sure it's super useful since it's just testing the same mocked values twice so you're probably fine with asserting the call count.